tiledb/sm/query/reader.cc

```diff
@@ -877,6 +880,43 @@ Status Reader::check_subarray() const {
   return Status::Ok();
 }
 
+Status Reader::check_validity_buffer_sizes() const {
+  // Verify that the validity buffer size for each
+  // nullable attribute is large enough to contain
+  // a validity value for each cell.
+  for (const auto& it : buffers_) {
+    const std::string& name = it.first;
+    if (array_schema_->is_nullable(name)) {
+      const uint64_t buffer_size = *it.second.buffer_size_;
+
+      uint64_t min_cell_num = 0;
+      if (array_schema_->var_size(name)) {
+        // todo arrow
```

Comment:
Were you trying to implement this in this PR?

Reply:
Oops, yes. Thanks. I just need to account for the possibility of an extra element. Good catch.